docs/deployment.rst

```diff
@@ -0,0 +1,36 @@
+Safeguard
+---------
+
+To balance contention between topics, certain safeguard-limits are placed on each topic-partition pull.
+These limits are configurable. Please see configuration section. Below is an further explanation
+for how these safeguards work, and why they are needed. By default, all the safeguard configurations
+are turned off, you need to configure them explicitly in your Camus configuration file.
+
+Limiting timestamp duration: This can be set with ``kafka.max.pull.hrs``.
+Since output data is partitioned by timestamp,
+large pulls may create a large number of files. For example, if you use hourly partition,
+a pull of 3 days data in a Camus job will simultaneously write to 72 files.
+Some versions of Hadoop don't behave well in this case.
+Therefore, the ability of limiting the timestamp duration is needed.
+The way how this works is that the timestamp for each event
+is compared to the timestamp of the first event.
+If the timestamp is greater than duration set in ``kafka.max.pull.hrs``,
+the pull is stopped and the next Camus job will pickup from where it left off.
+
+Limiting pull time for tasks: This can be set with ``kafka.max.pull.minutes.per.task``.
+if the task time is exceeded, the pulls are halted and the next Camus job will continue from the new offset.
+
+Limiting pull for historic data:  This can be set with ``kafka.max.historical.days``.
+Events older than a specified number of days will be ignored.
+This is useful in a situation where a valid offset is not available and
+a topic is getting pulled from the earliest offset.
+
+
+Exception Handling
```

Comment:
Let's remove this title. If we call this section as "Important Configuration Options", this section will just talk about one more config option.